src/app/g3w-ol/controls/streetviewcontrol.js

```diff
@@ -1,12 +1,74 @@
 import ApplicationState from 'core/applicationstate';
+const { base, inherit, XHR } = require('core/utils/utils');
 const utils = require('core/utils/ol');
-const InteractionControl = require('./interactioncontrol');
-const PickCoordinatesInteraction = require('../interactions/pickcoordinatesinteraction');
+const G3WObject = require('core/g3wobject');
+const GUI = require('gui/gui');
+const StreetViewComponent = require('gui/streetview/vue/streetview');
+const InteractionControl = require('g3w-ol/controls/interactioncontrol');
+const PickCoordinatesInteraction = require('g3w-ol/interactions/pickcoordinatesinteraction');
+
+const GoogleStreetViewApiUrl = `https://maps.googleapis.com/maps/api/`;
+
+function StreetviewService() {
+  this._position = null;
+  this.key = null; //get google key
+  this.errorKey = null;
+  this.setters = {
+    postRender(position) {}
+  };
+
+  this.init = function() {
+    return new Promise(async (resolve, reject) => {
+      this.key = ApplicationState.keys.vendorkeys.google;
+      if (this.key) {
+        try {
+          await XHR.get({
+            url: `${GoogleStreetViewApiUrl}streetview?location=0,0&size=456x456&key=${this.key}`
+          })
+        } catch(error) {
+          this.errorKey = error.responseText;
+          reject(error);
+        }
+      }
+      $script(`${GoogleStreetViewApiUrl}js?${this.key ? 'key=' + this.key : '' }`,
+        resolve
+      )
+    })
+  };
+
+  this.getKey = function(){
+    return this.key;
+  };
+
+  this.setKey = function(key) {
+    this.key = key;
+  };
+
+  this.getPosition = function() {
+    return this._position;
+  };
+
+  this.showStreetView = function(position) {
+    this._position = position;
+    GUI.setContent({
+      content: new StreetViewComponent({
+        service: this
+      }),
+      title: 'StreetView'
+    });
+  };
+
+  base(this);
+}
+
+inherit(StreetviewService, G3WObject);
+
+const streetviewService = new StreetviewService();
```

Comment:
@volterra79  can you include these features within a single class (`StreetviewService` --> `StreetViewControl`)

Reply:
Sure.